### DISCUSSION TITLE
Capabilities

### DISCUSSION THREAD

**Discussion by susnick** [editorconfig/editorconfig#527] [Q&A]
Im looking for a solution to the following problem. I have a coworker that uses prettier to convert his tabs to spaces with size 2. So if he had 3 tabs it would be 6 spaces. I prefer tabs with size of 4. Since all spaces have been committed in the project, can this convert spaces back to tabs. Currently only solutions I have found will take 4 spaces and make it a tab, but still leaves 2 spaces. I would think a solution would be to detect the number of spaces at the beginning of a line divided by the "from indent size" and insert that many tabs. Of course i don't want this trashing my git history so it needs to save back as spaces since that was originally used. This needs to only apply to viewing a file in my editor.

Is this possible or am I wasting my time? It seems like I need from -> to settings. Also need to mention once set, this needs to be automatic. no running commands anywhere.

**Comment by xuhdev** [comment]
Could you give an example? Overall it sounds like something that prettier can handle better.

**Reply by susnick** [reply]
Sure. I want this

```
if (true) {
  if (true) {
    if (true) {
      console.log("3 indents");
    }
  }
```

converted to this

```
if (true) {
 -> if (true) {
 ->  -> if (true) {
 ->  ->  -> console.log("3 indents");
 ->  -> }
 -> }
}
```

**Reply by xuhdev** [reply]
It's still unclear what you are asking. You seemed to have omitted a lot of details and we won't be able to understand your question. (e.g., what is `->` in your code examples above?)

Do you mean you would like to replace every 2 leading spaces on each line with a tab? You should be able to do so with `sed`: https://stackoverflow.com/questions/9222281/replace-leading-whitespace-with-sed-or-similar

**Reply by susnick** [reply]
-> is a tab. I tried examples in your link and none work. also it required me to run a command. that is a deal breaker for me. I primarily use vscode and the jetbrains editors, I expect to open a file and it shows the file as I want. Each occurrence of 2 spaces at the beginning of a line should be converted to a tab.

**Reply by xuhdev** [reply]
The link above shows you what direction you can look into by using `sed`. Why does using a command a deal breaker? `sed` is powerful and known to work well with this kind of line based processing. You can also try to find and replace two spaces with a tab in an editor and manually select out those that don't match your criteria. Or perhaps try a regex replacing. `prettier` or another formatter may also work if you don't mind them reformat the whole files by changing other parts, but you likely need to configure them a bit.

Anyway, I don't think this is something EditorConfig can help you with. The task sounds very specific to you and I would expect a little configuration/regex/scripting. Why not try to understand `sed` a bit and give it a try? It likely won't cost you too much time.
